src/lib/core.d.ts

```diff
@@ -1037,55 +1037,55 @@ interface ReadonlyArray<T> {
       * @param callbackfn A function that accepts up to three arguments. The every method calls the callbackfn function for each element in array1 until the callbackfn returns false, or until the end of the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
-    every(callbackfn: (value: T, index: number, array: ReadonlyArray<T>) => boolean, thisArg?: any): boolean;
+    every(callbackfn: (value: T, index: number, array: this) => boolean, thisArg?: any): boolean;
     /**
       * Determines whether the specified callback function returns true for any element of an array.
       * @param callbackfn A function that accepts up to three arguments. The some method calls the callbackfn function for each element in array1 until the callbackfn returns true, or until the end of the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
-    some(callbackfn: (value: T, index: number, array: ReadonlyArray<T>) => boolean, thisArg?: any): boolean;
+    some(callbackfn: (value: T, index: number, array: this) => boolean, thisArg?: any): boolean;

```

Comment:
Impact is very low because support for fully subclassing `Array` is very recent _and_ narrow (stable: only Edge). See Kangax:
http://kangax.github.io/compat-table/es6/#test-Array_is_subclassable_Array.prototype.filter

Only people who subclass `Array` _and_ override one of those methods with callbacks (`find`, `some`, etc.) are impacted.

But in the future you can expect several frameworks to "hack" arrays for their own special purposes... At which point the annotation will become a tiny bit relevant (only a tiny bit because I never see the third parameter used in practice).

I will also say that TS has had breaking changes by way of tightening its type checks or fixing incorrect declarations at several occasions in the past.

I'd say: now is the right time for the change -- otherwise never.